pkg/agents/team5/treaty.go

```diff
@@ -0,0 +1,160 @@
+		// Don't agree to treaty which is bounded by you having lower HP
+		// Don't agree to treaty which is bounded by the amount of food on platform being low
+		// Both of these are conditions where you will be desperate for food so shouldn't be limitting yourself if you do get the opurrtunity to eat
+		fallthrough
+	case (treaty.ConditionOp() == messages.GE || treaty.ConditionOp() == messages.GT) && treaty.Condition() == messages.Floor:
+		// Don't agree to treaty which is bounded by you being lower down in the tower for same as reasons above
+		fallthrough
+	case treaty.Condition() == messages.HP && treaty.ConditionValue() < a.HealthInfo().WeakLevel*2:
+		//Reject any HP condition based treaty if the condition is too strict, and you would be put into critical by not eating
+		fallthrough
+	case treaty.Condition() == messages.Floor && treaty.ConditionValue() != 1 && treaty.Duration() >= a.HealthInfo().MaxDayCritical:
+		//Reject any floor condition that involves us not being on the top floor and lasts for more than days you can survive in critical
+		//This is because there is a risk of signing your own death if you agree as you may be forced to eat no food with no get out condition
+		fallthrough
+	case treaty.Condition() == messages.AvailableFood && treaty.Request() == messages.LeaveAmountFood && treaty.ConditionValue()-treaty.RequestValue() < 3:
+		// Reject treaty in which you would not be able to eat enough food to avoid critical level
+		fallthrough
+	case treaty.Condition() == messages.AvailableFood && treaty.Request() == messages.LeavePercentFood && treaty.ConditionValue()*int(float64(100-treaty.RequestValue())/100) < 3:
+		// Reject treaty in which you would not be able to eat enough food to avoid critical level
+		a.RejectTreaty(msg)
+		return
+	}
+
+	//TODO: Develop the decision calculation.
+	//For now: Middle range of selfishness - selfishness + opinion of agent proposing treaty - duration of treaty/4
+	decision := 5 - a.selfishness + a.socialMemory[msg.SenderID()].favour - treaty.Duration()/4
+	if decision > 0 {
+		treaty.SignTreaty()
+		a.AddTreaty(treaty)
+		reply := msg.Reply(a.ID(), a.Floor(), msg.SenderFloor(), true)
+		a.SendMessage(reply)
+		a.Log("Accepted treaty", infra.Fields{"proposerID": msg.SenderID(), "proposerFloor": msg.SenderFloor(), "treatyID": msg.TreatyID()})
+		passingOnTo := a.Floor() + 1
+		if msg.SenderFloor() > a.Floor() {
+			passingOnTo = a.Floor() - 1
+		}
+		passItOn := messages.NewProposalMessage(a.ID(), a.Floor(), passingOnTo, treaty)
+		a.SendMessage(passItOn)
+	} else {
+		a.RejectTreaty(msg)
+	}
+}
+
+func (a *CustomAgent5) HandleTreatyResponse(msg messages.TreatyResponseMessage) {
+	if msg.Response() {
+		treaty := a.ActiveTreaties()[msg.TreatyID()]
+		a.ActiveTreaties()[msg.TreatyID()] = treaty
+		a.addToSocialFavour(msg.SenderID(), a.socialMemory[msg.SenderID()].favour+2)
+	}
+}
```

Comment:
I'm missing something or an agent could break your agent by sending a response message but modify the treaty id to one you don't have? xD `treaty := a.ActiveTreaties()[msg.TreatyID()]`

Just checking if the memory accesses in this function are always valid

Reply:
Please see 'Added check for active treaty in response handler' commit for fix